File: src/main/java/com/matamercer/microblog/models/entities/Follow.java
package com.matamercer.microblog.models.entities;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;

@Entity
@Table(name="follows")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Follow extends BaseModel{

    @ManyToOne
    @JoinColumn(name = "fk_follower_blog")
    private Blog follower;

    @ManyToOne
    @JoinColumn(name= "fk_followee_blog")
    private Blog followee;

    @Column
    private boolean notificationsEnabled;

    @Column
    private boolean muted;
}

File: src/main/java/com/matamercer/microblog/models/entities/Follow.kt
package com.matamercer.microblog.models.entities

import javax.persistence.*

@Entity
@Table(name = "follows", uniqueConstraints = [UniqueConstraint(columnNames = arrayOf("fk_follower_blog", "fk_followee_blog"))])
class Follow(
    @ManyToOne
    @JoinColumn(name = "fk_follower_blog")
    val follower: Blog? = null,

    @ManyToOne
    @JoinColumn(name = "fk_followee_blog")
    val followee: Blog? = null,

    @Column
    val notificationsEnabled: Boolean = false,

    @Column
    val muted: Boolean = false
): BaseModel(){}